### PR TITLE
Enhance the Category children End point

### LINKED ISSUE #29315: Enhance /api/v1/categories/children Endpoint to Return Parent List
### Parent Issue
#28493 

### User Story
As a developer, I need the `/api/v1/categories/children` endpoint to return the parentList of categories when `showChildrenCount: true` is sent. Alternatively, create a new parameter to handle this. This is necessary to provide the path to the parent category for deeper levels (2-N) to help users navigate back to the selected child category.

### Acceptance Criteria
- Modify the `/api/v1/categories/children` endpoint to include the parentList when `parentList: true` is sent.
- Alternatively, add a new parameter to request the parentList.
- Ensure the parentList provides the complete path to the parent category for deeper levels (2-N).
- Update the API documentation to reflect the changes.
- This is only of 1 Level, only for the children of the inode sent.
- Write unit tests to cover the new functionality.

### Proposed Objective
API Enhancement

### Proposed Priority
Priority 2 - Important

### dotCMS Version
master

### Problem

Screenshot 2024-07-19 at 5 16 05 PM

Request don't have parentList
Screenshot 2024-07-19 at 5 18 26 PM

When we sent filter query param, the endpoint retrieve parentList of the category
Screenshot 2024-07-19 at 5 18 49 PM

### Assumptions & Initiation Needs
Assuming the current implementation of the `/api/v1/categories/children` endpoint can be enhanced to include the parentList. Access to the current codebase and API documentation is required.

### Quality Assurance Notes & Workarounds
Quality assurance should verify that the parentList is returned correctly when requested and that the API remains stable. Ensure that edge cases, such as categories with no parents, are handled correctly.

### PR DESCRIPTION
**Pull request by freddyDOTCMS**
### Proposed Changes
* Add a new Query Parameter on the Endpoint to calculate the parentList

https://github.com/dotCMS/core/pull/29386/files#diff-1677f12a3b6c8909f3436afc7817819af2f46ac5ae6a37bdfab43690ffa24780R261

- Support all combinations of different parameters on the findAll factory method, now you can calculate the parent list and search just on one Category level on the same time for example

https://github.com/dotCMS/core/pull/29386/files#diff-bf828c747c99cefe73af7cec527db7f7e64bd4a66ba54a55eb1170f6a2996333R15-R990

This PR fixes: #29315

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java [hunk 1]
```diff
@@ -270,23 +268,20 @@ public final Response getChildren(@Context final HttpServletRequest httpRequest,
...
+        extraParams.put("parentList", parentList);
+        extraParams.put("showChildrenCount", showChildrenCount);
 
         try {
-            response = showChildrenCount == false ? this.paginationUtil.getPage(httpRequest, user, filter, page, perPage, orderBy,
-                    direction.equals("ASC") == true ? OrderDirection.ASC : OrderDirection.DESC, extraParams)
-                    : this.childrenCategoriesPaginationUtil.getPage(httpRequest, user, filter, page, perPage, orderBy,
-                            direction.equals("ASC") == true ? OrderDirection.ASC : OrderDirection.DESC, extraParams);
+            return  this.paginationUtil.getPage(httpRequest, user, filter, page, perPage, orderBy,
+                    direction.equals("ASC") == true ? OrderDirection.ASC : OrderDirection.DESC, extraParams);
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java:277]: `direction.equals()` already return `true`.

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryRecursiveQueryBuilder.java [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            "parentList_1", parentList_1,
+            "parentList_2", parentList_2,
+            "parentList_3", parentList_3,
+            "filterCategories", getFilterCategories(),
+            "orderBy",searchCriteria.orderBy,
+            "direction", searchCriteria.direction.toString()
+        ));
+    }
+
+    protected String getFilterCategories() {
```
Comment (victoralfaro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryRecursiveQueryBuilder.java:137]: IMHO this should be in a constant

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryQueryBuilder.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+ * This query is constructed using the {@link CategoryRecursiveQueryBuilder} class.
+ * Simple Query: This query traverses the category and tree tables. It is used when:
+ *
+ * You need to search across the entire tree.
+ * You want to search within a specific level, such as the children of a specific {@link Category} or the top-level categories.
+ * This query is constructed using the {@link CategorySimpleQueryBuilder} class.
+ */
+public abstract class CategoryQueryBuilder {
+
+    protected String rootInode;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryQueryBuilder.java:30]: set to final

**Review by jdotcms — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java [hunk 4]
```diff
@@ -635,9 +632,11 @@ private List<HierarchedCategory> convertForHierarchedCategories(final List<Map<S
 						final String parentsASJsonArray = "[" + row.get("path") + "]";
 						final List<ShortCategory> parentList = getShortCategories(parentsASJsonArray);
 						category.setParentList(parentList.subList(0, parentList.size() - 1));
-						category.setChildrenCount(ConversionUtils.toInt(row.get("childrencount"), 0));
 					}
 
+                    assert row != null;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java:637]: Remove this

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java (hunk 4 above)
Comment (freddyDOTCMS) [reply dotCMS/src/main/java/com/dotmarketing/portlets/categories/business/CategoryFactoryImpl.java:637]: Sonnar make me put it there

**Pull request merged**